### ISSUE TITLE
Implement Birch/MaxP variant

### ISSUE THREAD

**Issue by lintool** [castorini/pygaggle#88]
We should probably implement our own variant of Birch/BERT-MaxP in PyGaggle - 

+ Birch: https://www.aclweb.org/anthology/D19-1352/
+ BERT-MaxP: https://dl.acm.org/doi/10.1145/3331184.3331303

Let's call it `MonoBirchP` :)

**Comment by ronakice** [comment]
Sorry, I missed this, we're already doing MaxP and sum-based aggregation in Document Ranking. Birch is just a generalized weight-based sum aggregator that linearly interpolates with the BM25 scores right?

**Comment by lintool** [comment]
Correct. This is probably lower priority because we also have https://capreolus.ai/
